src/testoptix/CMakeLists.txt

```diff
@@ -1,37 +1,99 @@
+    if (VERBOSE)
+        message (STATUS "Current #defines are ${CURRENT_DEFINITIONS}")
+    endif ()
+    foreach (def ${CURRENT_DEFINITIONS})
+        set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} "-D${def}")
+    endforeach()
+    set (LLVM_COMPILE_FLAGS ${LLVM_COMPILE_FLAGS} ${SIMD_COMPILE_FLAGS} ${CSTD_FLAGS})
+
+    # Figure out what program we will use to make the bitcode.
+    if (NOT LLVM_BC_GENERATOR)
+        FIND_PROGRAM(LLVM_BC_GENERATOR NAMES "clang++" PATHS "${LLVM_DIRECTORY}/bin" NO_DEFAULT_PATH NO_CMAKE_SYSTEM_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH)
+    endif ()
+    # If that didn't work, look anywhere
+    if (NOT LLVM_BC_GENERATOR)
+        # Wasn't in their build, look anywhere
+        FIND_PROGRAM(LLVM_BC_GENERATOR NAMES clang++ llvm-g++)
+    endif ()
+
+    if (NOT LLVM_BC_GENERATOR)
+        message (FATAL_ERROR "You must have a valid llvm bitcode generator (clang++) somewhere.")
+    endif ()
+    if (VERBOSE)
+        message (STATUS "Using ${LLVM_BC_GENERATOR} to generate bitcode.")
+    endif()
+
+    # Command to turn the renderer library source file into LLVM bitcode .bc,
+    # then back into a C++ file with the bc embedded!
+    # This macro is set up to accept a .cu file, but could be adapted to accept
+    # a generic C++ file.
+    ADD_CUSTOM_COMMAND ( OUTPUT ${llvm_bc_cpp}
+      COMMAND ${LLVM_BC_GENERATOR}
+          "-I${OPTIX_INCLUDE_DIR}"
+          "-I${CUDA_INCLUDE_DIR}"
+          "-I${CMAKE_SOURCE_DIR}/cuda"
+          -D__CUDACC__ -D__CUDA_ARCH__=350
+          -O3 -Wno-format-security --cuda-gpu-arch=sm_35 --cuda-device-only -emit-llvm
+          ${llvm_src} -c -o ${llvm_bc}
+      COMMAND "${CMAKE_SOURCE_DIR}/src/liboslexec/serialize-bc.bash" ${llvm_bc} ${llvm_bc_cpp} "rend_llvm_compiled_ops"
+      MAIN_DEPENDENCY ${llvm_src}
+      DEPENDS "${CMAKE_SOURCE_DIR}/src/liboslexec/serialize-bc.bash"
+              ${llvm_src}
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/cuda" )
+ENDMACRO ()
+
 # Compile a CUDA file to PTX using NVCC.
 MACRO ( NVCC_COMPILE cuda_src ptxlist )
     GET_FILENAME_COMPONENT ( cuda_src_we ${cuda_src} NAME_WE )
-    SET ( cuda_ptx "${CMAKE_CURRENT_BINARY_DIR}/${cuda_src_we}.cu.ptx" )
+    SET ( cuda_ptx "${CMAKE_CURRENT_BINARY_DIR}/${cuda_src_we}.ptx" )
     SET ( ${ptxlist} ${${ptxlist}} ${cuda_ptx} )
 
     ADD_CUSTOM_COMMAND ( OUTPUT ${cuda_ptx}
         COMMAND ${CUDA_NVCC_EXECUTABLE}
-            -m64 -arch sm_35 -ptx --std=c++11 -O3 --use_fast_math
+            -DOSL_USE_FAST_MATH=1
+            -m64 -arch sm_35 -ptx --std=c++11 -O3 --use_fast_math --expt-relaxed-constexpr
             "-I${CUDA_INCLUDE_DIR}"
             "-I${OPTIX_INCLUDE_DIR}"
             "-I${CMAKE_SOURCE_DIR}/cuda"
+            "-I${OPENIMAGEIO_INCLUDE_DIR}"
+            "-I${CMAKE_BINARY_DIR}/include"
+            "-I${PROJECT_SOURCE_DIR}/src/include"
             ${cuda_src} -o ${cuda_ptx}
```

Comment:
I found that here, it also needed

    "-I${ILMBASE_INCLUDE_DIR}"

because the ImathVec_cuda.h includes some other Imath files and needs to know where to find them.